internal/client.go

```diff
@@ -626,6 +630,528 @@ type (
+	}
+
+	// ScheduleUpdate describes the desired new schedule from ScheduleHandle.Update.
+	ScheduleUpdate struct {
+		// Schedule - New schedule to replace the existing schedule with
+		Schedule *Schedule
+	}
+
+	// ScheduleUpdateOptions configure the parameters for updating a schedule.
+	ScheduleUpdateOptions struct{}
+
+	// ScheduleHandle represents a created schedule.
+	ScheduleHandle interface {
+		// GetID returns the schedule ID asssociated with this handle.
+		GetID() string
+
+		// Delete the Schedule 
+		Delete(ctx context.Context) error
+
+		// Backfill the schedule by going though the specified time periods and taking Actions as if that time passed by right now, all at once.
+		Backfill(ctx context.Context, backfill []ScheduleBackfill) error
+
+		// Update the Schedule. fn takes a description of the schedule and returns the new desired schedule.
+		// If update returns a nil response then no update will occur.
+		//
+		// NOTE: If two Update calls are made in parallel to the same Schedule there is the potential
+		// for a race condition.
+		Update(ctx context.Context, fn func(*ScheduleDescribeResponse) *ScheduleUpdate, options ScheduleUpdateOptions) error
+
+		// Describe fetches the Schedule's description from the Server
+		Describe(ctx context.Context) (*ScheduleDescribeResponse, error)
+
+		// Trigger an Action to be taken immediately. Will override the schedules default policy
+		// with the one specified here. If overlap is SCHEDULE_OVERLAP_POLICY_UNSPECIFIED the schedule
+		// policy will be used.
+		Trigger(ctx context.Context, overlap enumspb.ScheduleOverlapPolicy) error
+
+		// Pause the Schedule will also overwrite the Schedules current note with the new note.
+		// If an empty note is passed the Schedules note will be set to 'Paused via Go SDK'.
+		Pause(ctx context.Context, note string) error
+
+		// Unpause the Schedule will also overwrite the Schedules current note with the new note.
+		// If an empty note is passed the Schedules note will be set to 'Unpaused via Go SDK'.
+		Unpause(ctx context.Context, note string) error
+	}
+
+	// ScheduleActionResult describes when a schedule action took place
+	ScheduleActionResult struct {
+		// ScheduleTime - Time that the Action was scheduled for, including jitter.
+		ScheduleTime time.Time
+
+		// ActualTime - Time that the Action was actually taken.
+		ActualTime time.Time
+
+		// StartWorkflowResult - If action was ScheduleWorkflowAction, returns the
+		// ID of the workflow.
+		StartWorkflowResult ScheduleWorkflowExecution
+	}
+
+	// ScheduleListEntry
+	ScheduleListEntry struct {
+		// ScheduleID - The business identifier of the schedule.
+		ScheduleID string
+
+		// Spec - Describes when Actions should be taken.
+		Spec *ScheduleSpec
+
+		// Note - Informative human-readable message with contextual notes, e.g. the reason
+		// a Schedule is paused. The system may overwrite this message on certain
+		// conditions, e.g. when pause-on-failure happens.
+		Note string
```

Comment:
I have seen "note" in our Go code and "notes" in the proto API. Should we call this `Notes` everywhere instead? Why this mismatch?

Reply:
That is true. TS calls it `note` as well. I would prefer `note` because, generally, I think plurals should be used for lists. That being said I am in favor of changing to `notes` so we are consistent.

Reply:
I don't mind if we are consistent in our SDKs. It is unfortunate that we can't be consistent across the company, but oh well.